js/adapt-offlineStorage-xapi.js

```diff
@@ -0,0 +1,86 @@
+define([
+  'core/js/adapt',
+  './adapt-contrib-xapi'
+], function(Adapt, xapi) {
+
+  //xAPI handler for Adapt.offlineStorage interface.
+
+  // Use a lightweight fake model to pass into xAPI.sendState
+  var fakeModel = {
+    get: function() {
+      return 'offlineStorage';
+    }
+  };
+
+  var store = new Backbone.Model();
+  var isDataRestored = false;
+
+  Adapt.offlineStorage.initialize({
+
+    get: function(name) {
+      if (!name) {
+        return this.getAll();
+      }
+
+      if (!this.useTemporaryStore() && name.toLowerCase() === 'learnerinfo') {
+        return this.getLearnerInfo();
+      }
+
+      return store.get(name);
+    },
+
+    getAll: function() {
+      if (!isDataRestored) {
+        var state = xapi.get('state') || {};
+        store.set(state.offlineStorage);
+        isDataRestored = true;
+      }
+
+      //If not connected return just the store.
+      if (this.useTemporaryStore()) {
+        return store.toJSON();
+      }
+
+      return _.extend(store.toJSON(), {
+        learnerInfo: this.getLearnerInfo()
+      });
+    },
+
+    set: function(name, value) {
+      store.set(name, value);
+
+      // xAPI may not yet be initialised so use a soft trigger rather than hard calling xAPI.sendState
+      Adapt.trigger('state:change', fakeModel, store.toJSON());
+    },
+
+    useTemporaryStore: function() {
+      var config = Adapt.config.get('_xapi') || {};
+
+      return (!window.xapiWrapper || !config._isEnabled);
+    },
+
+    /**
+     * @returns {{name: string in the format Firstname Lastname, firstname: string, lastname: string }}
+     */
+    getLearnerInfo: function() {
+      var name = xapi.getLRSAttribute('actor').name;
+      var lastname;
+      var firstname;
+
+      if (name && name !== 'undefined' && name.indexOf(' ') > -1) {
+        var nameSplit = name.split(' ');
+        lastname = $.trim(nameSplit[0]);
```

Comment:
Are these the wrong way round?
Also, (minor): This doesn't offer complete coverage as names such as 'Luis van Gaal' will give firstname 'Luis', lastname 'van'

Reply:
I think this might be copied from the SCORM version where the name is stored in 'Lastname, Firstname' style

Reply:
Good spot @cahirodoherty-learningpool . You're right @moloko blatant rip.

Of course there's no way to do this perfectly do this ie no way to differentiate between multiple firstnames/lastnames (e.g Jamie Lee Curtis) but this will now take the first name as everything before the first space and the last name as everything after.

Reply:
> this will now take the first name as everything before the first space and the last name as everything after

Best you can really do given the data you've got to work with...